nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/test/java/org/apache/nifi/redis/service/TestRedisConnectionPoolService.java

```diff
@@ -19,17 +19,39 @@
 import org.apache.nifi.redis.RedisConnectionPool;
 import org.apache.nifi.redis.util.RedisUtils;
 import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.security.util.KeyStoreUtils;
+import org.apache.nifi.security.util.SslContextFactory;
+import org.apache.nifi.ssl.RestrictedSSLContextService;
+import org.apache.nifi.util.MockConfigurationContext;
+import org.apache.nifi.util.MockProcessContext;
+import org.apache.nifi.util.StandardProcessorTestRunner;
 import org.apache.nifi.util.TestRunner;
 import org.apache.nifi.util.TestRunners;
+import org.junit.Assert;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
 
 public class TestRedisConnectionPoolService {
 
+    public static final String SSL_CONTEXT_IDENTIFIER = "ssl-context-service";
     private TestRunner testRunner;
     private FakeRedisProcessor proc;
     private RedisConnectionPool redisService;
 
+    private static SSLContext sslContext;
+
+    @BeforeClass
+    public static void classSetup() throws IOException, GeneralSecurityException {
+        sslContext = SslContextFactory.createSslContext(KeyStoreUtils.createTlsConfigAndNewKeystoreTruststore());
```

Comment:
`KeyStoreUtils.createTlsConfigAndNewKeystoreTruststore()` create files in the `java.io.tmpdir` directory that need to be deleted after the test completes.  However, given that the unit test is only checking for the presence of certain SSLContext objects, this could be simplified to just using `SSLContext.getDefault()` and avoid the need to create the keystore and truststore files.
```suggestion
        sslContext = SSLContext.getDefault();
```